src/lib/homestore_backend/replication_state_machine.cpp

```diff
@@ -133,25 +137,31 @@ ReplicationStateMachine::get_blk_alloc_hints(sisl::blob const& header, uint32_t
     const ReplicationMessageHeader* msg_header = r_cast< const ReplicationMessageHeader* >(header.cbytes());
     switch (msg_header->msg_type) {
     case ReplicationMessageType::CREATE_SHARD_MSG: {
-        auto& pg_id = msg_header->pg_id;
+        pg_id_t pg_id = msg_header->pg_id;
         // check whether the pg exists
         if (!home_object_->pg_exists(pg_id)) {
             LOGI("can not find pg {} when getting blk_alloc_hint", pg_id);
             // TODO:: add error code to indicate the pg not found in homestore side
             return folly::makeUnexpected(homestore::ReplServiceError::NO_SPACE_LEFT);
         }
-        // Since chunks are selected when a pg is created, the chunkselector selects one of the chunks owned by the pg
+
+        auto v_chunkID = home_object_->resolve_v_chunk_id_from_msg(header);
+        RELEASE_ASSERT(v_chunkID.has_value(), "unkown chunk for create shard");
         homestore::blk_alloc_hints hints;
-        hints.pdev_id_hint = pg_id; // FIXME @Hooper: Temporary bypass using pdev_id_hint to represent
-                                    // pg_id_hint, "identical layout" will change it
+        // use pdev_id_hint (of type uint32_t) to store the values of pg_id and v_chunk_id.
+        // Both chunk_num_t and pg_id_t are of type uint16_t.
+        static_assert(std::is_same< pg_id_t, uint16_t >::value, "pg_id_t is not uint16_t");
+        static_assert(std::is_same< homestore::chunk_num_t, uint16_t >::value, "chunk_num_t is not uint16_t");
+        homestore::chunk_num_t v_chunk_id = v_chunkID.value();
+        hints.pdev_id_hint = ((uint32_t)pg_id << 16) | v_chunk_id;
```

Comment:
we will use application_hint here ,right?

Reply:
sure, done